collectors/framework/record_test.go

```diff
@@ -20,90 +20,180 @@ import (
 	"testing"
 
 	"github.com/dcos/dcos-metrics/producers"
+	"github.com/dcos/dcos-metrics/schema/metrics_schema"
+	"github.com/linkedin/goavro"
+	. "github.com/smartystreets/goconvey/convey"
 )
 
-func TestAvroRecordExtractDatapoint(t *testing.T) {
+var (
+	metricListNamespace = goavro.RecordEnclosingNamespace(metricsSchema.MetricListNamespace)
+	metricListSchema    = goavro.RecordSchema(metricsSchema.MetricListSchema)
+	datapointNamespace  = goavro.RecordEnclosingNamespace(metricsSchema.DatapointNamespace)
+	datapointSchema     = goavro.RecordSchema(metricsSchema.DatapointSchema)
+	tagNamespace        = goavro.RecordEnclosingNamespace(metricsSchema.TagNamespace)
+	tagSchema           = goavro.RecordSchema(metricsSchema.TagSchema)
+)
 
-	var (
-		testRecord = record{
-			Name: "dcos.metrics.Datapoint",
-			Fields: []field{
-				{
-					Name:  "test-name",
-					Datum: "name-field-test",
-				},
-				{
-					Name:  "test-value",
-					Datum: "value-field-test",
-				},
-				{
-					Name:  "test-unit",
-					Datum: "unit-field-test",
-				},
+var (
+	testDatapoint = record{
+		Name: "dcos.metrics.Datapoint",
+		Fields: []field{
```

Comment:
Note; had to change the ordering of the fields in this test to match the generated schema.